### PR TITLE
PT-158827776 Add first batch of new HTTP endpoints

### PR DESCRIPTION
**Pull request by jur0**
The PR adds endpoints:
```
/blocks/top
/key-blocks/*
/micro-blocks/*
/generations/*
```
We still keep the old endpoints. We plan to create follow up PRs for:
* adding of the rest of new API enpoints
* cleaning up old endpoints

### CODE REVIEW

File: apps/aehttp/src/aehttp_api_parser.erl [hunk 1]
```diff
@@ -42,10 +48,10 @@
                                     micro ->
                                         {ok, HMap0} = aec_headers:serialize_to_map(aec_blocks:to_header(Block)),
 %% currently the pending endpoint can return a pending microblock. The
```
Comment (velzevur) [inline apps/aehttp/src/aehttp_api_parser.erl]: Since the `GetPendingBlock` is changed to `GetPendingKeyBlock` - the endpoint is no longer returning a micro block. We are safe to remove this comment and the check bellow for block signatures being `undefined`

File: apps/aehttp/src/aehttp_dispatch_ext.erl [hunk 2]
```diff
@@ -59,7 +171,7 @@ handle_request('GetBlockLatest', Req, _Context) ->
 handle_request('GetBlockPending', Req, _Context) ->
```
Comment (velzevur) [inline apps/aehttp/src/aehttp_dispatch_ext.erl]: Since we have a `GetBlockPending` do we really need this anymore?

File: apps/aehttp/src/aehttp_api_parser.erl (hunk 1 above)
Comment (velzevur) [reply apps/aehttp/src/aehttp_api_parser.erl]: `GetBlockPending` is still around, if its removal will be a subject to another PR - so shall this comment and the workaround

File: apps/aehttp/src/aehttp_dispatch_ext.erl [hunk 3]
```diff
@@ -59,7 +171,7 @@ handle_request('GetBlockLatest', Req, _Context) ->
...
     get_block(fun aehttp_logic:get_block_pending/0, Req, json, false);
 
-handle_request('GetKeyBlockByHeight', Req, _Context) ->
+handle_request('GetKeyBlockByHeightObsolete', Req, _Context) ->
```
Comment (velzevur) [inline apps/aehttp/src/aehttp_dispatch_ext.erl]: This is a duplicated endpoint. Will its removal be a subject to another PR? (I strongly suspect the path `/block/height/{height}` is being used by Python tests so its removal will require some changes there)

File: config/swagger.yaml [hunk 4]
```diff
@@ -37,22 +37,355 @@ paths:
...
-          description: successful operation
+          description: 'Successful operation'
           schema:
             $ref: '#/definitions/Top'
       security:
+  /key-blocks/current:
+    get:
+      tags:
+        - external
+        - gossip
```
Comment (velzevur) [inline config/swagger.yaml]: As discussed - we need to address endpoint tags. Tracked in [Revisit swagger tags](https://www.pivotaltracker.com/story/show/158872647)

File: apps/aehttp/src/aehttp_dispatch_ext.erl (hunk 3 above)
Comment (jur0) [reply apps/aehttp/src/aehttp_dispatch_ext.erl]: This will be removed in separate PR.

File: config/swagger.yaml [hunk 5]
```diff
@@ -37,6 +37,22 @@ paths:
 ## External APIs definitions
 ###
 
+  /top:
```
Comment (fabiankrol) [inline config/swagger.yaml]: I'm reintroducing `/top` endpoint for full backward compatibility. It'll be removed with other legacy endpoints

Comment (fabiankrol) [reply config/swagger.yaml]: @jur0 @velzevur ^

Comment (velzevur) [reply config/swagger.yaml]: I see @jur0 had fixed the python UATs, do we still need it?
